src/System.Runtime.Environment/src/OSName.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace System.Runtime.InteropServices
+{
+    public struct OSName : IEquatable<OSName>
+    {
+        private string _name;
+
+        private const string WindowsName = "windows";
+        private const string LinuxName = "linux";
+        private const string OSXName = "osx";
+
+        private static readonly OSName s_windows = new OSName(WindowsName);
```

Comment:
I think it would be better to just create new instance in the getter. Since this is a struct we will get a copy anyway.